src/aiida_workgraph/engine/workgraph.py

```diff
@@ -252,22 +267,22 @@ def on_wait(self, awaitables: t.Sequence[t.Awaitable]):
 
     def _build_process_label(self) -> str:
         """Use the workgraph name as the process label."""
-        return f"WorkGraph<{self.inputs.wg['name']}>"
+        return f"WorkGraph<{self.inputs.workgraph_data['name']}>"
 
     def on_create(self) -> None:
         """Called when a Process is created."""
         from aiida_workgraph.utils.analysis import WorkGraphSaver
 
         super().on_create()
-        wgdata = self.inputs.wg._dict
-        restart_process = (
-            orm.load_node(wgdata["restart_process"].value)
-            if wgdata.get("restart_process")
-            else None
+        self.node.label = self._raw_inputs["workgraph_data"]["name"]
```

Comment:
Shouldn't we use here the `WorkGraphSpec.WORKGRAPH_DATA_KEY` instead of `workgraph_data`? Because later you use this to pass it  WorkGraphEngine. So if the spec key changes this will fail.

Reply:
yes.